common/crypto/platform/openssl/crypto.cpp

```diff
@@ -21,11 +21,20 @@
 #include <vector>
 #include <memory>
 
+
+#ifdef MENDER_CRYPTO_OPENSSL_LEGACY
+#include <openssl/engine.h>
+#else
+#include <openssl/provider.h>
+#include <openssl/store.h>
+#endif // MENDER_CRYPTO_OPENSSL_LEGACY
```

Comment:
These (the above) don't seem to be required for the code below. At least not for the new one.